usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateNoOverlappingStopTimeInTripBlockTest.java

```diff
@@ -0,0 +1,1523 @@
+        when(firstTripFirstStopTime.getStopSequence()).thenReturn(10);
+
+        final StopTime firstTripLastStopTime = mock(StopTime.class);
+        when(firstTripLastStopTime.getArrivalTime()).thenReturn(820);
+        when(firstTripLastStopTime.getDepartureTime()).thenReturn(1000);
+        when(firstTripLastStopTime.getStopSequence()).thenReturn(11);
+
+        mockFirstTripStopTimeCollection.put(10, firstTripFirstStopTime);
+        mockFirstTripStopTimeCollection.put(11, firstTripLastStopTime);
+
+        final SortedMap<Integer, StopTime> mockSecondTripStopTimeCollection = new TreeMap<>();
+
+        final StopTime secondTripFirstStopTime = mock(StopTime.class);
+        when(secondTripFirstStopTime.getArrivalTime()).thenReturn(1723);
+        when(secondTripFirstStopTime.getDepartureTime()).thenReturn(1808);
+
+        final StopTime secondTripLastStopTime = mock(StopTime.class);
+        when(secondTripLastStopTime.getArrivalTime()).thenReturn(1820);
+        when(secondTripLastStopTime.getDepartureTime()).thenReturn(1857);
+
+        mockSecondTripStopTimeCollection.put(12, secondTripFirstStopTime);
+        mockSecondTripStopTimeCollection.put(13, secondTripLastStopTime);
+
+        final SortedMap<Integer, StopTime> mockThirdTripStopTimeCollection = new TreeMap<>();
+        final StopTime thirdTripFirstStopTime = mock(StopTime.class);
+        when(thirdTripFirstStopTime.getArrivalTime()).thenReturn(940);
+        when(thirdTripFirstStopTime.getDepartureTime()).thenReturn(1216);
+        when(thirdTripFirstStopTime.getStopSequence()).thenReturn(14);
+
+        final StopTime thirdTripLastStopTime = mock(StopTime.class);
+        when(thirdTripLastStopTime.getArrivalTime()).thenReturn(1308);
+        when(thirdTripLastStopTime.getDepartureTime()).thenReturn(1350);
+        when(thirdTripLastStopTime.getStopSequence()).thenReturn(15);
+
+        mockThirdTripStopTimeCollection.put(14, thirdTripFirstStopTime);
+        mockThirdTripStopTimeCollection.put(15, thirdTripLastStopTime);
+
+        when(mockDataRepo.getAllTripByBlockId()).thenReturn(mockTripPerBlockIdCollection);
+        when(mockDataRepo.getStopTimeByTripId("2")).thenReturn(mockFirstTripStopTimeCollection);
+        when(mockDataRepo.getStopTimeByTripId("5")).thenReturn(mockSecondTripStopTimeCollection);
+        when(mockDataRepo.getStopTimeByTripId("8")).thenReturn(mockThirdTripStopTimeCollection);
+
+        final Map<String, Calendar> mockCalendarCollection = new HashMap<>();
+        final Calendar calendarForServiceA = mock(Calendar.class);
+        final Calendar calendarForServiceB = mock(Calendar.class);
+        final Calendar calendarForServiceC = mock(Calendar.class);
+        mockCalendarCollection.put("a", calendarForServiceA);
+        mockCalendarCollection.put("b", calendarForServiceB);
+        mockCalendarCollection.put("c", calendarForServiceC);
+        when(calendarForServiceA.isOverlapping(calendarForServiceB)).thenReturn(false);
```

Comment:
Same on not mocking domain class methods

Reply:
As discussed, implementation of domain layer classes is tested separately, so that tests of use case layer only verify interactions with mocked objects.